pkg/wallet/wallet.go

```diff
@@ -56,7 +59,7 @@ func NewWallet(location string) (*Wallet, error) {
 
 // NewWalletFromFile creates a Wallet from the given wallet file path.
 func NewWalletFromFile(path string) (*Wallet, error) {
-	file, err := os.OpenFile(path, os.O_RDWR, os.ModeAppend)
+	file, err := os.Open(path)
```

Comment:
Why do we keep this handle in `Wallet` at all? We read whole file, we don't really care about it afterwards until there is a need to update it (in `writeRaw`).